Pretty.fs

```diff
@@ -37,10 +40,38 @@ let indent level = new string(' ', level * 4)
 /// Enters a new line at the given indent level.
 let lnIndent level = "\n" + indent level
 
+/// Helpers for turning a Doc into a Styled
+/// for syntax highlighting keywords, literals, identifiers and view syntax
+/// respectively
+let syntax d = Styled([Magenta], d)
+let syntaxLiteral d = Styled([Blue], d)
+let syntaxIdent d = Styled([Cyan], d)
+let syntaxView d = Styled([Yellow], d)
+
+/// Magic
+let stylise s d = 
+    let colCode =
+        function
+        | Black -> 0
+        | Red -> 1
+        | Green -> 2
+        | Yellow -> 3
+        | Blue -> 4
+        | Magenta -> 5
+        | Cyan -> 6 
+        | White -> 7
+
+    let code c = (30 + colCode c).ToString ()
+
+    let prefix = "\u001b[" + (String.concat ";" <| List.map code s) + "m"
+    let suffix = "\u001b[0m"
```

Comment:
Well anything higher level would be a wrapper around ANSI escape codes or similar, terminal technologies really don't have. For example, outputting raw ANSI escape codes is how vim can do syntax highlighting.

Either way it's not really that important, I just have them to help me visualise which bits of the source program relate to the outputted AST and metadata (although i cleaned that bit up before this commit)